DigitalLearningSolutions.Data/Models/DelegateUpload/DelegateTableRow.cs

```diff
@@ -0,0 +1,156 @@
+        public string? Answer3 { get; set; }
+        public string? Answer4 { get; set; }
+        public string? Answer5 { get; set; }
+        public string? Answer6 { get; set; }
+        public string? AliasId { get; set; }
+        public string? Email { get; set; }
+
+        public BulkUploadResult.ErrorReason? Error { get; set; }
+        public RowStatus RowStatus { get; set; }
+
+        public bool Validate(IEnumerable<int> allowedJobGroupIds)
+        {
+            if (!JobGroupId.HasValue || !allowedJobGroupIds.Contains(JobGroupId.Value))
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidJobGroupId;
+            }
+
+            if (string.IsNullOrEmpty(LastName))
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidLastName;
+            }
+
+            if (string.IsNullOrEmpty(FirstName))
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidFirstName;
+            }
+
+            if (string.IsNullOrEmpty(Email))
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidEmail;
+            }
+
+            if (!Active.HasValue)
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidActive;
+            }
+
+            return !Error.HasValue;
+        }
+
+        public bool MatchesDelegateUser(DelegateUser delegateUser)
+        {
+            if (CandidateNumber != null && (delegateUser.AliasId ?? string.Empty) != AliasId)
+            {
+                return false;
+            }
+
+            if ((delegateUser.FirstName ?? string.Empty) != FirstName)
+            {
+                return false;
+            }
+
+            if (delegateUser.LastName != LastName)
+            {
+                return false;
+            }
+
+            if (delegateUser.JobGroupId != JobGroupId!.Value)
+            {
+                return false;
+            }
+
+            if (delegateUser.Active != Active!.Value)
+            {
+                return false;
+            }
+
+            if ((delegateUser.Answer1 ?? string.Empty) != Answer1)
+            {
+                return false;
+            }
+
+            if ((delegateUser.Answer2 ?? string.Empty) != Answer2)
+            {
+                return false;
+            }
+
+            if ((delegateUser.Answer3 ?? string.Empty) != Answer3)
+            {
+                return false;
+            }
+
+            if ((delegateUser.Answer4 ?? string.Empty) != Answer4)
+            {
+                return false;
+            }
+
+            if ((delegateUser.Answer5 ?? string.Empty) != Answer5)
+            {
+                return false;
+            }
+
+            if ((delegateUser.Answer6 ?? string.Empty) != Answer6)
+            {
+                return false;
+            }
+
+            if ((delegateUser.EmailAddress ?? string.Empty) != Email)
+            {
+                return false;
+            }
+
+            return true;

```

Comment:
L148-153 can be a simple return (I think we fixed before but it's reverted back.